packages/synapse-react-client/src/lib/containers/discussion_forum/DiscussionReply.tsx

```diff
@@ -52,75 +51,69 @@ export const DiscussionReply: React.FC<DiscussionReplyProps> = ({
       ) : (
         <>
           {message && (
-            <div>
-              <UserCard
-                withAvatar={true}
-                avatarSize="MEDIUM"
-                showCardOnHover={true}
-                size={SMALL_USER_CARD}
-                ownerId={reply.createdBy}
-              />
-              <div className="message-body">
-                <MarkdownSynapse
-                  markdown={message}
-                  objectType={ObjectType.REPLY}
+            <>
+              <div>
+                <UserCard
+                  withAvatar={true}
+                  avatarSize="MEDIUM"
+                  showCardOnHover={true}
+                  size={SMALL_USER_CARD}
+                  ownerId={reply.createdBy}
                 />
-                <span>
-                  posted {formatDate(dayjs(reply.createdOn), 'M/D/YYYY')}
-                </span>
-                <div style={{ float: 'right' }}>
-                  <button onClick={() => onClickLink()}>
-                    <IconSvg icon="link" />
-                  </button>
-                  {isCurrentUserAuthor && (
-                    <button onClick={() => setShowReplyModal(true)}>
-                      <IconSvg icon="edit" />
-                    </button>
-                  )}
-                  {entityBundle?.permissions.canModerate && (
-                    <button onClick={() => setShowDeleteModal(true)}>
-                      <IconSvg icon="delete" />
+                <div className="message-body">
+                  <MarkdownSynapse
+                    markdown={message}
+                    objectType={ObjectType.REPLY}
+                  />
+                  <span>
+                    posted {formatDate(dayjs(reply.createdOn), 'M/D/YYYY')}
+                  </span>
+                  <div style={{ float: 'right' }}>
+                    <button onClick={() => onClickLink()}>
+                      <IconSvg icon="link" />
                     </button>
-                  )}
+                    {isCurrentUserAuthor && (
+                      <button onClick={() => setShowReplyModal(true)}>
+                        <IconSvg icon="edit" />
+                      </button>
+                    )}
+                    {entityBundle?.permissions.canModerate && (
+                      <button onClick={() => setShowDeleteModal(true)}>
+                        <IconSvg icon="delete" />
+                      </button>
+                    )}
+                  </div>
                 </div>
               </div>
-            </div>
+              <ForumThreadEditor
```

Comment:
Use the refactored ForumThreadEditor to display the dialog -- 
![feature_DiscussionReply_EditReply](https://user-images.githubusercontent.com/26949006/236354988-5189c07c-7abb-49f2-8772-7181fb51442a.png)